xcode/Subconscious/Shared/Components/Common/Transclude/TranscludeBlockLayoutFragment.swift

```diff
@@ -0,0 +1,158 @@
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+import SwiftUI
+
+#if os(iOS)
+import UIKit
+#else
+import Cocoa
+#endif
+import CoreGraphics
+
+struct EmbeddedTranscludePreview: View {
+    var label: String = "Subconscious"
+    
+    var body: some View {
+        VStack(alignment: .leading) {
+            Text("title")
+            Text(label)
+                .multilineTextAlignment(.leading)
+                .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .topLeading)
+        }
+        .padding(4)
+        .frame(maxWidth: .infinity, maxHeight: 128) // TODO: share constant for all instances of 128
+        .background(.white)
+        .overlay(
+            RoundedRectangle(cornerRadius: 4)
+                .stroke(.purple, lineWidth: 2)
+            )
+        .padding(2)
+    }
+}
+
+extension UIView {
+    func asImage() -> UIImage {
+        let renderer = UIGraphicsImageRenderer(bounds: bounds)
+        let img = renderer.image { rendererContext in
+            rendererContext.cgContext.saveGState()
+            
+            // SwiftUI views render upside down by default, flip them
+            let flipVertical: CGAffineTransform = CGAffineTransform(a: 1, b: 0, c: 0, d: -1, tx: 0, ty: bounds.height )
+            rendererContext.cgContext.concatenate(flipVertical)
+            
+            layer.render(in: rendererContext.cgContext)
+            
+            rendererContext.cgContext.restoreGState()
+        }
+        
+        return img
+    }
+}
+
+class TranscludeBlockLayoutFragment: NSTextLayoutFragment {
+    // Max height constraint
+    let SLASHLINK_PREVIEW_HEIGHT = 128.0
+    var text: String?
+    
+    override var leadingPadding: CGFloat { return 10 }
+    override var trailingPadding: CGFloat { return 10 }
+    override var topMargin: CGFloat { return 0 }
+    override var bottomMargin: CGFloat { return 0 }
+    
+    private var img: UIImage?
+    
+    private func render() {
+        let v = EmbeddedTranscludePreview(label: text ?? "Testing")
+        // Host our SwiftUI view within a UIKit view
+        let hosted = UIHostingController(rootView: v)
+        guard let view = hosted.view else {
+            return
+        }
+        
+        view.translatesAutoresizingMaskIntoConstraints = false
+        
+        // We have to mount the view before it will actually do layout calculations
+        UIApplication.shared.windows.first?.rootViewController?.view.addSubview(hosted.view)
```

Comment:
Meant to call attention to this line before, without mounting the `UIHostingController` the view cannot be rendered into an image but... this line is gross and yields the a warning: `'windows' was deprecated in iOS 15.0: Use UIWindowScene.windows on a relevant window scene instead`

I don't know what a "relevant window scene" refers to from this context.